### PR TITLE
Add Devcontainer files and Docs

### PR DESCRIPTION
**Pull request by maxbartel**
When working inside the onnx-mlir development docker container, changes one does have to be pushed to some branch or they will be deleted. To solve this problem, one could also mount the source folder to a container. This way changes are saved on the local machine. 

To support this, VS Code has a feature called [Devcontainers](https://code.visualstudio.com/docs/devcontainers/containers). They expose a nice API with many useful features like scripts that always run when attaching to a container.

This PR brings in example files to use a devcontainer. I leveraged the available `onnxmlirczar/onnx-mlir-llvm-static:amd64` image which already has a prebuilt LLVM version.

### CODE REVIEW

**Review by AlexandreEichenberger — COMMENTED** [review]
Appears to be fine; @gongsu832 @caoimhinuibrian do you want to chime in?

**Comment by maxbartel** [comment]
@AlexandreEichenberger @gongsu832 @caoimhinuibrian Could I get reviews for this please?

File: docs/devcontainer-example/Dockerfile.llvm-project [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# By default, use ubuntu:focal
+ARG BASE_IMAGE="onnxmlirczar/onnx-mlir-llvm-static:amd64"
```
Comment (gongsu832) [inline docs/devcontainer-example/Dockerfile.llvm-project:2]: Change the comment above or remove it.

The Docker.md probably should mention that things like Mac M1/2 will require virtualization such as Rosetta, which is still under development (https://github.com/docker/roadmap/issues/384), to run amd64 containers. Qemu can also work but will be pretty slow.

**Review by AlexandreEichenberger — COMMENTED** [review]
I will defer my judgement to @gongsu832 as he is our expert on build, CI, docker, and many other things...

**Comment by maxbartel** [comment]
@gongsu832 thank you for the review! I mention the M1 limitation now.

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Review by gongsu832 — APPROVED** [review]
LGTM

**Comment by maxbartel** [comment]
@AlexandreEichenberger I don't know if @caoimhinuibrian wants to jump in. If not, could you merge this please if the pipeline passes?

**Pull request merged**